### PR TITLE
feat(build): migrate to GHA

### PR DESCRIPTION
**Pull request by florin139**

### CODE REVIEW

File: .github/workflows/commit-lint.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          fetch-depth: 0
+      - name: Dry-run release
+        uses: tradeshift/actions-semantic-release@v1
+        id: semantic-release
+        with:
+          branches: |
+            [
+              'master'
+            ]
```
Comment (wejendorp) [inline .github/workflows/commit-lint.yml:18]: We actually don't need this field any more, this should be the default. I'm removing it where I remember :grimacing:

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/checkout@v2
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: "npm"
+
+      - name: 📥 Download deps
+        run: npm install
```
Comment (wejendorp) [inline .github/workflows/main.yml:29]: This should be `npm ci`

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/checkout@v2
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: 📥 Download deps
+        run: npm install
```
Comment (wejendorp) [inline .github/workflows/main.yml:53]: This should be `npm ci`

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+      - name: 🚀 Release
+        uses: tradeshift/actions-semantic-release@v1
+        id: semantic-release
+        with:
+          registry: https://npm.pkg.github.com/
+          npm_publish: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (wejendorp) [inline .github/workflows/main.yml:68]: Ah, we need to use the public registry for this public package.

https://www.npmjs.com/package/http-mockserver

See https://github.com/Tradeshift/eslint-config-tradeshift/blob/58b999f5/.github/workflows/main.yml#L63-L79 for example.

Comment (florin139) [reply .github/workflows/main.yml:68]: good catch :)

**Review by cristianmarinescu — APPROVED** [review]

**Pull request merged**